tests/jobs/test_scheduler_job.py

```diff
@@ -1669,22 +1668,24 @@ def test_scheduler_keeps_scheduling_pool_full(self, dag_maker):
         with dag_maker(
             dag_id='test_scheduler_keeps_scheduling_pool_full_d1',
             start_date=DEFAULT_DATE,
-        ) as dag_d1:
+        ):
             BashOperator(
                 task_id='test_scheduler_keeps_scheduling_pool_full_t1',
                 pool='test_scheduler_keeps_scheduling_pool_full_p1',
                 bash_command='echo hi',
             )
+        dag_d1 = dag_maker.dag
```

Comment:
I thought  the dag_maker yielded the dag, so this style of change shouldn't be necessary?

Reply:
For some reason if you use `dag_maker` more than once, the `as` variables are binded to the same DAG:

```python
with dag_maker(...) as dag1:
    ...

with dag_maker(...) as dag2:
    ...

assert dag1 is dag2. # True!
```

I don't understand how this could happen, maybe there's some Pytest magic involved, but this is the easiest workaround anyway so I just did it instead of going down the rabbit hole.